db/sync.go

```diff
@@ -61,11 +65,11 @@ func (db *DB) BeginSync(batchPushURL string, diffServerURL string, diffServerAut
 	// Pull
 	headSnapshot, err := baseSnapshot(db.noms, head)
 	if err != nil {
-		return hash.Hash{}, syncInfo, fmt.Errorf("sync failed: could not find head snapshot: %w", err)
+		return hash.Hash{}, syncInfo, fmt.Errorf("sync %s failed: could not find head snapshot: %w", syncInfo.SyncID, err)
```

Comment:
[cbb?] A way to reduce the boilerplate here would be to wrap the error at a higher level  (e.g., inside one of the two top-level sync calls) and add the syncID there. That way we can be sure that any errors coming up through that path are so-tagged.

Reply:
will do, thanks for the suggestion